### PR TITLE
Log appender improvements

### PR DESCRIPTION
**Pull request by sfackler**
This PR contains a series of 3 separate cleanups to our internal logging infrastructure. It's probably best to review each separately. This will improve the performance of stdout logging as well as prepare for the implementation of #85.

### CODE REVIEW

File: witchcraft-server/src/logging/logger/rolling_file.rs [hunk 1]
```diff
@@ -11,44 +11,53 @@
...
 use tokio::fs::{self, File, OpenOptions};
-use tokio::io::{self, AsyncWrite, AsyncWriteExt, BufWriter};
+use tokio::io::{self, AsyncWrite, AsyncWriteExt};
 use tokio::task;
 
 const MAX_LOG_SIZE: u64 = 1024 * 1024 * 1024;
 
 struct CurrentFile {
-    file: BufWriter<File>,
+    file: BufBytesSink<FileBytesSink>,
```
Comment (TheKeveloper) [inline witchcraft-server/src/logging/logger/rolling_file.rs]: nit: `sink` or `bytesSink`?

File: witchcraft-server/src/logging/logger/rolling_file.rs [hunk 2]
```diff
@@ -80,7 +89,7 @@ impl RollingFileAppender {
         let len = file.metadata().await.map_err(Error::internal_safe)?.len();
 
         let archive_locator = ArchiveLocator::new(name);
-        let date = Utc::now().date();
+        let date = Utc::now().date_naive();
```
Comment (TheKeveloper) [inline witchcraft-server/src/logging/logger/rolling_file.rs:92]: Why the change to `NaiveDate`?

Comment (sfackler) [reply witchcraft-server/src/logging/logger/rolling_file.rs:92]: `Date` is now deprecated: https://docs.rs/chrono/latest/chrono/struct.Date.html

File: witchcraft-server/src/logging/logger/rolling_file.rs (hunk 1 above)
Comment (sfackler) [reply witchcraft-server/src/logging/logger/rolling_file.rs]: Fixed!

**Review by TheKeveloper — APPROVED** [review]
